cmd/up/deploy.go

```diff
@@ -0,0 +1,194 @@
+// Copyright 2025 The Okteto Authors
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package up
+
+import (
+	"context"
+	"errors"
+	"os"
+	"time"
+
+	"github.com/okteto/okteto/cmd/deploy"
+	deployCMD "github.com/okteto/okteto/cmd/deploy"
+	pipelineCMD "github.com/okteto/okteto/cmd/pipeline"
+	"github.com/okteto/okteto/cmd/utils"
+	"github.com/okteto/okteto/pkg/analytics"
+	"github.com/okteto/okteto/pkg/cmd/pipeline"
+	"github.com/okteto/okteto/pkg/constants"
+	"github.com/okteto/okteto/pkg/env"
+	oktetoErrors "github.com/okteto/okteto/pkg/errors"
+	oktetoLog "github.com/okteto/okteto/pkg/log"
+	"github.com/okteto/okteto/pkg/log/io"
+	"github.com/okteto/okteto/pkg/model"
+	"github.com/okteto/okteto/pkg/okteto"
+	"github.com/spf13/afero"
+	"k8s.io/client-go/kubernetes"
+)
+
+const (
+	// OktetoAutoDeployEnvVar if set the application will be deployed while running okteto up
+	OktetoAutoDeployEnvVar = "OKTETO_AUTODEPLOY"
+)
+
+// devEnvDeployerManager deploys the dev environment
+type devEnvDeployerManager struct {
```

Comment:
As we discussed offline, there are a couple of things to change an improve:
* Reduce the number of dependencies in the structs, as some of them are parameters of deploy function, that properties of the struct
* Remove the indirection of `devEnvEnvDeployStrategy`. It is not needed, as we can have a command provider function as part of devEnvDeployerManager, which would allow you to test everything, reducing the complexity of these structs